File: scheduler-server/src/schemas/address_book.py
"""
这里定义通讯录接口的响应结构。

通讯录接口会把实例、Runtime Target、以及群组视图一起返回，
方便前端左侧导航一次拿全。
"""
from pydantic import BaseModel


class AddressBookAgent(BaseModel):
    id: int
    agent_key: str
    cs_id: str
    display_name: str
    role_name: str | None
    enabled: bool


class AddressBookInstance(BaseModel):
    id: int
    name: str
    status: str
    agents: list[AddressBookAgent]


class AddressBookRuntimeTarget(BaseModel):
    id: int
    runtime_type: str
    display_name: str
    role_name: str | None
    cs_id: str
    instance_name: str
    enabled: bool


class AddressBookGroupMember(BaseModel):
    id: int
    instance_id: int | None = None
    agent_id: int | None = None
    runtime_target_id: int | None = None
    display_name: str
    agent_key: str | None = None
    instance_name: str | None = None
    runtime_type: str | None = None
    cs_id: str | None = None


class AddressBookGroup(BaseModel):
    id: int
    name: str
    description: str | None
    members: list[AddressBookGroupMember]


class AddressBookResponse(BaseModel):
    instances: list[AddressBookInstance]
    runtime_targets: list[AddressBookRuntimeTarget] = []
    groups: list[AddressBookGroup]

File: web-client/src/types/api/addressBook.ts
/**
 * 通讯录接口的原始响应类型。
 *
 * 这一层保持与后端字段一致。
 */
export interface AddressBookAgentResponse {
    id: number;
    agent_key: string;
    cs_id: string;
    display_name: string;
    role_name: string | null;
    enabled: boolean;
}

export interface AddressBookInstanceResponse {
    id: number;
    name: string;
    status: string;
    agents: AddressBookAgentResponse[];
}

export interface AddressBookRuntimeTargetResponse {
    id: number;
    runtime_type: string;
    display_name: string;
    role_name: string | null;
    cs_id: string;
    instance_name: string;
    enabled: boolean;
}

export interface AddressBookGroupMemberResponse {
    id: number;
    instance_id: number | null;
    agent_id: number | null;
    runtime_target_id: number | null;
    display_name: string;
    agent_key: string | null;
    instance_name: string | null;
    runtime_type: string | null;
    cs_id: string | null;
}

export interface AddressBookGroupResponse {
    id: number;
    name: string;
    description: string | null;
    members: AddressBookGroupMemberResponse[];
}

export interface AddressBookResponse {
    instances: AddressBookInstanceResponse[];
    runtime_targets: AddressBookRuntimeTargetResponse[];
    groups: AddressBookGroupResponse[];
}
